portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/pages/reset/admin-initiated-password-reset.js

```diff
@@ -0,0 +1,18 @@
+
+
+function getPasscode(env) {
+    try {
+        var passcode = callOSGiService("org.wso2.is.portal.user.client.api.RecoveryMgtService", "persistPasscode",
+            ["user1"]);
+    } catch (e) {
+        return {errorMessage: 'Error while generating the passcode'};
+    }
+    return passcode;
+}
+
+function onPost(env){
+    var passcode = getPasscode(env);
+    sendToClient("passCode", passcode);
```

Comment:
getPasscode(env) might return an errorMessage. Need to handle if (passcode.errorMessage) case and give front end alert if it is true.